MaxText/layers/linears.py

```diff
@@ -471,24 +471,35 @@ def load_balance_loss(self, top_k_indices, logits):
     return loss
 
   def dense_matmul(self, inputs, gate_logits, w0_kernel, w1_kernel, wo_kernel):
+    gate_logits = nn.with_logical_constraint(gate_logits, ("activation_batch", None, None))
```

Comment:
Nit: "activation_batch", "activation_length", "activation_embed"

Reply:
Also include "activation_embed" below where possible, I think some of None correspond to capacity axis which is fine to leave none, but for embed prefer to use explicit "activation_embed" over None

Reply:
sounds good! Let me cross check after adding them.